snd-server/spotify_crawler.py

```diff
@@ -1,203 +1,129 @@
+            track_id=track_id
+        )
 
-                    if isinstance(extra_features, Ok):
-                        try:
-                            track = track.dict()
-                            track.update(extra_features.value.dict())
-                            tracks.append(SoundationsTrackWithFeatures.parse_obj(track))
-                        except:
-                            print("Error while parsing track")
+        match track_features_result:
+            case Ok(result):
+                return result
 
-        return tracks
+            case Err(err):
+                raise HTTPException(status_code=err.http_code, detail=err.message)
 
-    def fetch_albums_by_artist(
+    async def __fetch_artist_tracks(
         self, artist: SpotifyApiArtist
-    ) -> Result[List[SpotifyApiAlbum], str]:
-        artist_id = artist.id
-
-        url = self.artists_album_url.format(id=artist_id, limit=10, offset=0)
-        albums = self.request(url, SpotifyApiAlbumSearchResponse)
-
-        if isinstance(albums, Ok):
-            return Ok(albums.value.albums)
-        else:
-            return Err("Error while parsing album")
-
-    def fetch_tracks_by_genre(self, genre: str) -> List[SoundationsTrackWithFeatures]:
-        tracks = []
-        for i in range(0, self.config.items_per_search, self.config.spotify_limit):
-            url = self.track_by_genre_url.format(
-                genre=genre, limit=self.config.spotify_limit, offset=i
+    ) -> List[SoundationsTrackWithFeatures]:
+        tracks: list[SoundationsTrackWithFeatures] = []
+        for i in range(
+            0, self.ctx.config.items_per_search, self.ctx.config.spotify_limit
+        ):
+            tracks_search_result = await self.ctx.spotify_api.search_tracks_by_artist(
+                artist=artist, limit=self.ctx.config.spotify_limit, offset=i
             )
-            tracks_general = self.request(url, SoundationsRawTrackResponse)
-
-            if isinstance(tracks_general, Ok):
-                tracks_general = tracks_general.value.tracks
-                for track in tracks_general:
-                    extra_features = self.fetch_track_features(track.id)
 
-                    if isinstance(extra_features, Ok):
-                        try:
-                            track = track.dict()
-                            track.update(extra_features.value.dict())
-                            tracks.append(SoundationsTrackWithFeatures.parse_obj(track))
-                        except:
-                            print("Error while parsing track")
+            match tracks_search_result:
+                case Ok(result):
+                    for raw_track in result.items:
+                        track_with_features = await self.__fetch_track_features(
+                            raw_track.id
+                        )
+
+                        track = SoundationsTrackWithFeatures(
```

Comment:
```suggestion
                        track_with_features = SoundationsTrackWithFeatures(
```

this one is actually track with features :)